var/spack/repos/builtin/packages/camp/package.py

```diff
@@ -10,10 +10,8 @@
 
 def hip_repair_options(options, spec):
     # there is only one dir like this, but the version component is unknown
-    options.append(
-        "-DHIP_CLANG_INCLUDE_PATH="
-        + glob.glob("{}/lib/clang/*/include".format(spec["llvm-amdgpu"].prefix))[0]
-    )
+    path = glob.glob("{}/lib/clang/*/include".format(spec["llvm-amdgpu"].prefix))[0]
+    options.append(cmake_cache_path("HIP_CLANG_INCLUDE_PATH", path))
```

Comment:
This will break all CmakePackage instances, the fix from #32504 is more correct, but probably ought to be put into `hip_repair_options_cache` or similar so we can still reuse it.

Reply:
Agreed - the changes in this branch will break things. I will remove those changes from this PR and use fixes in those branches.